python/tank/deploy/tank_commands/unregister_folders.py

```diff
@@ -0,0 +1,311 @@
+        Action.__init__(self, 
+                        "unregister_folders", 
+                        Action.TK_INSTANCE, 
+                        ("Unregisters the folders for an object in Shotgun."), 
+                        "Admin")
+        
+        # this method can be executed via the API
+        self.supports_api = True
+
+        self.parameters = {}
+        
+        self.parameters["path"] = { "description": "Path to unregister. Any child paths will be unregistered too.",
+                                    "default": None,
+                                    "type": "str" }
+        
+        self.parameters["entity"] = { "description": ("Entity to unregister. Should be a Shotgun style entity "
+                                                      "dictionary with keys type and id."),
+                                      "default": None,
+                                      "type": "dict" }
+        
+    
+    def run_interactive(self, log, args):
+        """
+        Tank command accessor
+        """
+        
+        if not self.tk.pipeline_configuration.get_shotgun_path_cache_enabled():            
+            # remote cache not turned on for this project
+            log.error("Looks like this project doesn't synchronize its folders with Shotgun! "
+                      "If you want to turn on synchronization for this project, run "
+                      "the 'upgrade_folders' tank command.")
+            return
+        
+        # there are multiple syntaxes for this command
+        # - context based: tank Shot ABC123 unregister_folders
+        # - path based: tank unregister_folders path1, path2, path3
+        
+        if self.context.entity:
+            # context based - e.g. 'tank Shot ABC123 unregister_folders' 
+            self._unregister_entity(self.context.entity, log, prompt=True)
+        
+        else:
+            # no context - so it is path based instead 
+            if len(args) == 0:
+                # display help message
+                
+                log.info("This command is used to unregister folders that have been created via the Toolkit "
+                         "folder creation. When applications are launched and folders are created, new entries "
+                         "are added to the Shotgun FilesystemLocation table. These are then used to track the "
+                         "relationship between Shotgun objects and folders on disk. If you ever need to "
+                         "undo these associations, you can use this command. ")
+                log.info("")
+                log.info("You can use it by passing a Shotgun object, e.g.")
+                log.info("> tank Shot ABC123 unregister_folders")
+                log.info("")
+                log.info("Alternatively, you can use it by passing one or more paths, e.g.")
+                log.info("> tank unregister_folders /path/to/folder_a /path/to/folder_b")
+                log.info("")
```

Comment:
It's a little weird that with one syntax you provide the entity info before `unregister_folders` and the other you provide path info *after* `unregister_folders`. But I see no other better way and the syntax for the path-based workflow is pretty simple and intuitive.